### DISCUSSION TITLE
Why Unsqueezed() X?

### DISCUSSION THREAD

**Discussion by shmuhammadd** [mrdbourke/pytorch-deep-learning#211] [Q&A]
Hi, why do we need to unsequeeze() this In chapter2?

`X = torch.arange(start, end, step)#.unsqueeze(dim=1)`

<img width="484" alt="image" src="https://user-images.githubusercontent.com/38854463/206905794-39ab57b0-ec75-4a8a-9434-46152ce134ae.png">

**Comment by mrdbourke** [comment]
Hi @shmuhammad2004,

`unsqueeze()` removes all single dimensions from a tensor.

Have you tried running the code with and without `unsqueeze()`? What happens to the inputs and outputs?
